daprdocs/content/en/operations/components/pluggable-components/pluggable-components-overview.md

```diff
@@ -0,0 +1,122 @@
+| State Store |  `state`   |  [state.proto]   |  [Redis](https://github.com/dapr/components-contrib/tree/master/state/redis)   | [concept]({{<ref "state-management-overview">}}), [howto]({{<ref "howto-get-save-state">}}), [api spec]({{<ref "state_api">}})                                        |
+|   Pub/sub   |  `pubsub`  |  [pubsub.proto]  |  [Redis](https://github.com/dapr/components-contrib/tree/master/pubsub/redis)  | [concept]({{<ref "pubsub-overview">}}), [howto]({{<ref "howto-publish-subscribe">}}), [api spec]({{<ref "pubsub_api">}})                                              |
+|  Bindings   | `bindings` | [bindings.proto] | [Kafka](https://github.com/dapr/components-contrib/tree/master/bindings/kafka) | [concept]({{<ref "bindings-overview">}}), [input howto]({{<ref "howto-triggers">}}), [output howto]({{<ref "howto-bindings">}}), [api spec]({{<ref "bindings_api">}}) |
+
+2. **Create service scaffolding.** Use the [protocol buffers and gRPC tools](https://grpc.io) to create the scaffolding for the service. You may want to get acquainted with [the gRPC concepts documentation](https://grpc.io/docs/what-is-grpc/core-concepts/).
+
+As an example, below is a snippet from the protocol buffer definition file service that defines the gRPC service definition for pluggable component state stores ([state.proto]).
+
+```protobuf
+// StateStore service provides a gRPC interface for state store components.
+service StateStore {
+  // Initializes the state store component with the given metadata.
+  rpc Init(InitRequest) returns (InitResponse) {}
+  // Returns a list of implemented state store features.
+  rpc Features(FeaturesRequest) returns (FeaturesResponse) {}
+  // Deletes the specified key from the state store.
+  rpc Delete(DeleteRequest) returns (DeleteResponse) {}
+  // Get data from the given key.
+  rpc Get(GetRequest) returns (GetResponse) {}
+  // Sets the value of the specified key.
+  rpc Set(SetRequest) returns (SetResponse) {}
+  // Ping the state store. Used for liveness purposes.
+  rpc Ping(PingRequest) returns (PingResponse) {}
+
+  // Deletes many keys at once.
+  rpc BulkDelete(BulkDeleteRequest) returns (BulkDeleteResponse) {}
+  // Retrieves many keys at once.
+  rpc BulkGet(BulkGetRequest) returns (BulkGetResponse) {}
+  // Set the value of many keys at once.
+  rpc BulkSet(BulkSetRequest) returns (BulkSetResponse) {}
+}
+```
+
+The interface for the `StateStore` service exposes 9 methods:
+
+- 2 methods for initialization and components capability advertisement (Init and Features)
+- 1 method for health-ness or liveness check (Ping)
+- 3 methods for CRUD (Get, Set, Delete)
+- 3 methods for bulk CRUD operations (BulkGet, BulkSet, BulkDelete)
+
```

Comment:
```suggestion

2. **Create service scaffolding.** Use [protocol buffers and gRPC tools](https://grpc.io) to create the necessary scaffolding for the service. You may want to get acquainted with [the gRPC concepts documentation](https://grpc.io/docs/what-is-grpc/core-concepts/).

The tools can generate code targeting [any gRPC-supported language](https://grpc.io/docs/what-is-grpc/introduction/#protocol-buffer-versions). This code will serve as the base for your server and it provide functionality to handle client calls along with infrastructure to decode incoming requests, execute service methods, and encode service responses.

The generated code is not complete. It will be missing concrete implementation for the methods your target service defines, i.e., the core of your pluggable component. This is further explored in the next item. Additionally, you will also have to provide code on how to handle Unix Socket Domain integration, which is Dapr specific, and code handling integration with your downstream services.
```